crates/viewer/re_ui/src/syntax_highlighting.rs

```diff
@@ -24,26 +23,65 @@ pub trait SyntaxHighlighting {
 // ----------------------------------------------------------------------------
 
 /// Easily build syntax-highlighted text.
-pub struct SyntaxHighlightedBuilder {
-    pub style: Arc<Style>,
+pub struct SyntaxHighlightedBuilder<'a> {
+    pub style: &'a Style,
+    pub tokens: &'a DesignTokens,
     pub job: LayoutJob,
 }
 
-/// Easilut build
-impl SyntaxHighlightedBuilder {
-    pub fn new(style: Arc<Style>) -> Self {
+/// Easily build syntax-highlighted [`LayoutJob`]s.
+impl<'a> SyntaxHighlightedBuilder<'a> {
+    pub fn new(style: &'a Style, tokens: &'a DesignTokens) -> Self {
         Self {
             style,
+            tokens,
             job: LayoutJob::default(),
         }
     }
 
     #[inline]
     pub fn append(mut self, portion: &dyn SyntaxHighlighting) -> Self {
-        portion.syntax_highlight_into(&self.style, &mut self.job);
+        portion.syntax_highlight_into(self.style, &mut self.job);
         self
     }
 
+    /// Some string data. Will be quoted.
+    pub fn code_string_value(&mut self, portion: &str) {
+        let mut format = monospace_text_format(self.style);
+        format.color = self.tokens.code_string;
+        self.job.append("\"", 0.0, format.clone());
+        self.job.append(portion, 0.0, format.clone());
+        self.job.append("\"", 0.0, format);
+    }
+
+    /// A string name (e.g. the key of a map).
+    pub fn code_name(&mut self, portion: &str) {
```

Comment:
How is this different from just a string?

Should this be called "identifier" instead?